src/store/ethereum/index.ts

```diff
@@ -184,9 +187,18 @@ async function setProvider(context: ActionContext, p: IWalletProvider) {
   const address = await signer.getAddress()
   context.state.signer = signer
   context.state.address = address
+  context.state.wallet = p.provider || null
   ethereumModule.commitSetWalletNetworkId(p.chainId)
 }
 
+async function onLogout(context: ActionContext) {
+  // yep. Now this module knows about provider inner workings...
+  if (context.state.walletType === "walletconnect") {
```

Comment:
Can you add a disconnect method to `IWalletProvider` and call that here instead of checking for wallet type?

Reply:
Sure